src/client/components/MealById.js

```diff
@@ -0,0 +1,158 @@
+import React, { useContext, useState } from "react";
+import Header from "./Header";
+import Footer from "./Footer";
+import { useParams } from "react-router-dom";
+import DataContext from "./DataContext";
+import { Link } from "react-router-dom";
+
+export default function MealById() {
+  const { newMeals, reviews } = useContext(DataContext);
+  const { id } = useParams();
+  const [inputState, setInputstate] = useState(false);
+  const [name, setName] = useState("");
+  const [phoneNumber, setPhoneNumber] = useState("");
+  const [numberOfGuests, setNumberOfGuests] = useState("");
+  const [email, setEmail] = useState("");
+
+  function newReservation() {
+    {
+      if (email !== "") {
+        setInputstate(true);
+        fetch("/api/reservations", {
+          method: "POST",
+          headers: { "Content-Type": "application/json" },
+          body: JSON.stringify({
+            contact_name: name,
+            contact_phonenumber: phoneNumber,
+            contact_email: email,
+            meal_id: id,
+            number_of_guests: numberOfGuests,
+          }),
+        })
+          .catch((e) => {
+            setError(e);
+            alert("please enter the Valid details");
+          })
+          .finally(() => {
+            setInputstate(false);
+            alert("Your Reservation has been saved. Enjoy the Experience.");
+          });
+      }
+    }
+  }
+  const mealById = newMeals.find((meal) => meal.id === parseInt(id));
+  console.log({mealById});
+  const reviewsById = reviews.filter(
+    (review) => review.meal_id === parseInt(id)
+  );
+  function reviewStars(stars) {
+    if (parseInt(stars) === 1) {
+      return <span>⭐</span>;
+    } else if (parseInt(stars) === 2) {
+      return <span>⭐⭐</span>;
+    } else if (parseInt(stars) === 3) {
+      return <span>⭐⭐⭐</span>;
+    } else if (parseInt(stars) === 4) {
+      return <span>⭐⭐⭐⭐</span>;
+    } else if (parseInt(stars) === 5) {
+      return <span>⭐⭐⭐⭐⭐</span>;
+    }
+  }
```

Comment:
```suggestion
  function reviewStars(stars) {
    const starsToRender = Array(stars).fill(0); // Creates an array with as many items as stars, and puts a 0 in each slot so we can iterate over it
    return (
      <span>
        {starsToRender.map(() => ⭐)}
      </span>
    );
  }
```

With this code you can support basically any number of stars. Code that includes that many if branches is usually a sign to re-think your data model!